src/push/monitor.ts

```diff
@@ -148,6 +151,15 @@ export async function buildMonitorSchema(monitors: Monitor[], isV2: boolean) {
       monitor.setContent(content);
       Object.assign(schema, { content, filter });
     }
+    const size = monitor.size();
+    const sizeKB = Math.round(size / 1000);
+    if (sizeKB > SIZE_LIMIT_KB) {
+      let outer = bold(`Aborted: Bundled code ${sizeKB}kB exceeds the recommended ${SIZE_LIMIT_KB}kB limit. Please check the dependencies imported.\n`);
+      const inner = `* ${config.id} - ${source.file}:${source.line}:${source.column}\n`;
+      outer += indent(inner);
+      throw red(outer);
```

Comment:
It might be better DX if we iterate through all monitors and collect all errors before pushing so that we report all at once, instead of just the first, wdyt?

Reply:
I thought about it, But it might cause the errors to be not meaningful if user decides to pull in a dep like `playwright-core` etc which would get included across all journeys if it was in a helper library (common case in ecp). Throwing it away first avoids those issues. This is why we also kept it this way from the beginning.